ledger/ledger-on-posix-filesystem/src/main/scala/com/daml/ledger/on/filesystem/posix/StateKeys.scala

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.on.filesystem.posix
+
+import java.nio.file.Path
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.DamlStateKey
+
+object StateKeys {
+  private val NameChunkSize: Int = 32
+
+  def resolveStateKey(root: Path, key: DamlStateKey): Path = {
```

Comment:
This needs to always add an extra chunk for the file, or we could end up with collisions of the form `a/b/c` colliding with `a/b/c/d`. Appending `/state` or something would fix this (it needs to be something non-hex so it can't collide itself).

Reply:
Turns out this isn't an issue because variable-length strings seem to be encoded with a length prefix, which makes prefix collisions impossible.